src/SwitchTransition.js

```diff
@@ -8,8 +8,7 @@ function areChildrenDifferent(oldChildren, newChildren) {
   if (
     React.isValidElement(oldChildren) &&
     React.isValidElement(newChildren) &&
-    oldChildren.key != null &&
-    oldChildren.key === newChildren.key
+    ((oldChildren.key == null && newChildren.key == null) || oldChildren.key === newChildren.key)
```

Comment:
Why would we want to treat `null` and `undefined` the same here?

Reply:
@taion, OK. This is how I initially coded it but decided it would be better handled in the component. I'll close the fork therefore.